src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,286 @@
+        self.operators = [0] * (num_operands - 1) # Creating a list for operators according to number of operands
+
+        MAXNUM = _constants.max
+        MINNUM = _constants.min
+        if (bit): # Setting minimum and maximum according to bitwise or arithmetic calculations
+            MAXNUM = _constants.bitmax
+            MINNUM = _constants.bitmin
+
+        # Generating the expression
+        for i in range(num_operands - 1):
+            self.operands[i] = random.randint(MINNUM, MAXNUM)
+            self.operators[i] = random.randint(0, len(self.operators_list) - 1)
+        self.operands[num_operands - 1] = random.randint(MINNUM, MAXNUM)
+
+    # Arithmetic operations
+
+    # Addition
+    def add(self, op1, op2):
+        sum = op1 + op2
+        if sum < _constants.min: # Minimum / maximum number to safely add to operand 1
+            op2 = _constants.min - random.randint(_constants.min, op1)
+            sum = op1 + op2
+        elif sum > _constants.max:
+            op2 = _constants.max - random.randint(op1, _constants.max)
+            sum = op1 + op2
+        return {'oper': '+', 'op2': op2, 'sum': sum}
+
+    # Subtraction
+    def sub(self, op1, op2):
+        sum = op1 - op2
+        if sum < _constants.min: # Minimum / maximum number to safely subtract from operand 1
+            op2 = - (_constants.min - random.randint(_constants.min, op1))
+            sum = op1 - op2
+        elif sum > _constants.max:
+            op2 = - (_constants.max - random.randint(op1, _constants.max))
+            sum = op1 - op2
+        return {'oper': '-', 'op2': op2, 'sum': sum}
+
+    # Multiplication
+    def mul(self, op1, op2):
+        sum = op1 * op2
+        if sum < _constants.min: # Minimum / maximum number to safely multiply operand 1
+            op2 = _constants.min / op1
+        elif sum > _constants.max:
+            op2 = _constants.max / op1
+        op2 = int(op2) # Rounding because it might not be an integer
+        sum = op1 * op2
+        return {'oper': '*', 'op2': op2, 'sum': sum}
+
+    # Division (floor division)
+    def div(self, op1, op2):
+        if op2 == 0: # To avoid division by zero
+            if (random.randint(0,1)):
+                op2 = random.randint(1, _constants.max)
+            else:
+                op2 = random.randint(_constants.min, -1)
+        sum = op1 / op2
+        sum = int(sum) # Rounding because it might not be an integer
+        return {'oper': '/', 'op2': op2, 'sum': sum}
+
+    # Modulus
+    # Difference between % operator in python and js
+    # in python: -xxx % yy => yy - (abs(xxx) % yy)
+    # in js: -xxx % yy => - abs(xxx) % yy
+    def mod(self, op1, op2):
+        if op2 == 0:
+            if (random.randint(0,1)):
+                op2 = random.randint(1, _constants.max)
+            else:
+                op2 = random.randint(_constants.min, -1)
+        if (((op1 < 0 and op2 > 0) or (op1 > 0 and op2 < 0)) and (op1 % op2 != 0)):
```

Comment:
`((op1 < 0 and op2 > 0) or (op1 > 0 and op2 < 0))`

This check means that `op1` and `op2` signs differ, so
`op1 * op2 < 0` results the same ( `+ * - = -`,  `- * + = -`)